### ISSUE TITLE
Doesn't work on Shield TV

### ISSUE THREAD

**Issue by ersan** [brodfueh/SpectrumTV_webview#1]
Tried this on my two Fire TV devices and it worked fine.  On Shield TV it gets stuck on 'Scanning MiniGuide'

On the Fire TV I noticed the miniguide would cycle through all the channels I have on that screen, but on the Shield TV it just stays on the first page of channels and nothing happens.

Pressing right or up also crashes the app.  I had to revert to the version this was forked from for it to work again unfortunately, though this one is much better.

**Comment by brodfueh** [comment]
Hi,
Thanks for trying the app. I think I found the bug. I don't have a Shield TV device but recreated a similar error using a Google TV virtual device in Android Studio.   I added a new level spectv-v2.0.1.apk with the fix. Could you give it a test. I recommend you do a clean install, ie uninstall the old level.   https://github.com/brodfueh/SpectrumTV_webview/releases/download/Spectv-V2/spectv-v2.0.1.apk

If it works I'll merge it into the release

Thanks

**Comment by ersan** [comment]
Just tested it out and it works!  Thank you.

It did crash the first couple times I pressed right to open the mini guide but after the third launch it seems to be working fine.

**Comment by brodfueh** [comment]
Glad it worked. Looking into the miniguide crash. Found a few possible causes.  
1) On initial install the app defaults to channel 1,  if channel 1 is not in the lineup a crash could occur.
2) pressing right arrow before the miniguide scan has completed, "Scanning MiniGuide" message is still being displayed
3) The miniguide scan failed 

You think any of these possibilities match what you saw?  Also, when the app starts it shows a message indicating how many channels found  "Found <nnn> channels".  Does the number displayed look correct, based on your Spectrum plan?

Thanks

**Comment by ersan** [comment]
Back again, the same exact issue - not cycling through the miniguide and getting stuck on first open with 'Scanning MiniGuide' in the top right - is happening again on my third Shield TV - the first two worked fine somehow.

1. There is no channel 1 in my lineup, it automatically tunes to channel 13 when I open the app.
2. Haven't tried that.
3. Now when it gets stuck on scanning it says "MiniGuide NOT AVAILABLE" when I press right, or "Guide NOT AVAILABLE" when I press up on the remote.

**Comment by brodfueh** [comment]
Thanks for testing.  I picked up a MECOOL Android 11 box last week and was able to recreate the problem you described.   Just finished testing this morning on MECOOL.  You can download updated ark from here. https://github.com/brodfueh/SpectrumTV_webview/releases/download/Spectv-V2/spectv-v2.0.1-a.apk

Haven't updated the code on github yet. Want to do some more testing on other devices.

Thanks

**Comment by ersan** [comment]
That fixed it, thanks again!

**Comment by brodfueh** [comment]
Thats great.  Got a few questions. Just want to make sure the scan is working correctly.
1) Did the miniguide sorted correctly by channels number?
2) Did the number of channels found matches your Spectrum plan?

just fyi, working on new feature to auto process channel number input after about 2,5 seconds, removes the requirement to hit enter/dpad center to process the channel.  Should be available in a week or 2.

Thanks

**Comment by ersan** [comment]
1. Yes the guide is sorted by channel number.
2. Not sure, I started counting a couple times and eventually lost track but there are at least 300 on the website and the app says 354 which seems right.

SHIELD remote doesn't have number keys so I can't really help or test with that, but I'm sure others will appreciate it.

Personally the only thing I'd really like is a button to turn on and off closed captioning, and maybe an option to filter by favorited channels only.

**Comment by brodfueh** [comment]
Hi,
Looked at adding closed captioning, prototyped  the function using the arrow down button to toggle off/on, when on the video view.  Adding more options is a bit more difficult, probably need to add some kinda of setup page.  

I'll be updating the code base in a few days to add the auto process channel number and CC support.

Thanks again for testing

**Comment by ersan** [comment]
You won't believe it, happening again with another new Shield TV I am trying to set up.

**Comment by brodfueh** [comment]
Hi, I fit a problem with scanning a few weeks ago and uploaded a new package https://github.com/brodfueh/SpectrumTV_webview/releases/download/Spectv-v2.0.2-beta/spectv-v2.0.2-beta.apk

Haven't made it the default yet. Hopefully  it fixes the problem your having. 

Let me know,
Thanks

**Comment by ersan** [comment]
Unfortunately that was the version I was already using and it does have the issue.

**Comment by brodfueh** [comment]
Ok, I'll see if I can find the issue. Thinking about a redesign for the scan, to many issue with the current design

**Comment by brodfueh** [comment]
Hi,
I uploaded a new release apk https://github.com/brodfueh/SpectrumTV_webview/releases/download/Spectv-v2.0.3/spectv-v2.0.3.apk

redesigned the scan function, hopefully it fixes the problem

**Comment by ersan** [comment]
Yep, working again thank you.  Here's hoping it sticks this time.

**Comment by brodfueh** [comment]
good , will be updating the code on github. thanks for testing

**Comment by brodfueh** [comment]
level 2.0.3 merged into master

**Comment by bluelakefish** [comment]
@brodfueh - Just discovered your excellent work.  Do you have any plans for adding a left arrow to change last/previous channel?

**Comment by brodfueh** [comment]
Hi, just saw your post, thanks for trying out the app.  Adding left arrow to change last/previous channel looks like a good idea. Looks like a small code update. Trying to decide if it would be useful when viewing the miniguide also. Currently left arrow on miniguide acts the same as up arrow. Might be nice to jump to last/previous channel.

I'll update the issue when I have a test version with the support.

**Comment by bluelakefish** [comment]
Thanks for the reply!  Appreciate your work!

**Comment by bluelakefish** [comment]
Just updated the app. Works perfectly, thank you.
